lib/core/src/display/shape/text/text_field/word_occurrence.rs

```diff
@@ -0,0 +1,202 @@
+//! This module contains definitions for word occurrences in TextField.
+
+use crate::prelude::*;
+
+use std::ops::Range;
+
+use crate::display::shape::text::text_field::content::TextFieldContent;
+use crate::display::shape::text::text_field::cursor::Cursor;
+use data::text::TextLocation;
+
+
+
+// =================
+// === WordRange ===
+// =================
+
+/// A struct containing indices for word's start and end.
+#[derive(Shrinkwrap,Debug,Clone)]
+pub struct WordRange {
+    /// A property containing the word start and end.
+    #[shrinkwrap(main_field)]
+    pub word_range : Range<TextLocation>,
+    is_selected    : bool
+}
+
+impl WordRange {
+    /// Creates a new `WordRange`.
+    pub fn new(word_range : Range<TextLocation>) -> Self {
+        let is_selected = false;
+        Self {word_range,is_selected}
+    }
+
+    /// Returns a word under cursor, if any.
+    pub fn word_at_cursor(content:&TextFieldContent, cursor:&Cursor) -> Option<Self> {
+        let line  = cursor.position.line;
+        let chars = content.lines()[line].chars();
+        if let Some(range) = get_index_range_of_word_at(chars,cursor.position.column) {
+            let column = range.start;
+            let start  = TextLocation {line,column};
+            let column = range.end;
+            let end    = TextLocation {line,column};
+            Some(Self::new(start..end))
+        } else {
+            None
+        }
+    }
+}
+
+
+
+// =======================
+// === WordOccurrences ===
+// =======================
+
+/// A struct containing word occurrences in `TextFieldContent`.
```

Comment:
There should be explanation what is the definition of word. Perfectly if it will be on each structure referring to "words" (and function in TextField which uses this structure), or at least at this module root - but then each structure should refer to this docs (e.g. _for definition of word, see `word_occurrence` module doc_.